models/models.js

```diff
@@ -45,3 +44,38 @@ exports.updateReview = (review_id, updates) => {
       return response.rows[0];
     });
 };
+
+exports.selectAllReviews = (
+  category,
+  sort_by = "created_at",
+  order_by = "DESC"
+) => {
+  const validSortColumns = ["created_at", "review_id", "title"];
+  const validOrder = ["ASC", "DESC"];
+
+  let queryStr = `SELECT reviews.*, COUNT(comments.review_id) AS comment_count FROM reviews LEFT JOIN comments ON reviews.review_id=comments.review_id`;
+
+  queryValues = [];
+  if (category) {
+    queryStr += ` WHERE category = $1 GROUP BY reviews.review_id`;
+    queryValues.push(category);
+  } else {
+    queryStr += ` GROUP BY reviews.review_id`;
+  }
+
+  if (!validSortColumns.includes(sort_by) || !validOrder.includes(order_by)) {
```

Comment:
66-67 you don't NEED to add this error handler manually.  If something other than ASC or DESC is used, postgres will throw an error itself - you can use the psql error code to send an error.